internal/olm/client/client.go

```diff
@@ -214,7 +216,53 @@ func (c Client) DoCSVWait(ctx context.Context, key types.NamespacedName) error {
 		return curPhase == olmapiv1alpha1.CSVPhaseSucceeded, nil
 	}
 
-	return wait.PollImmediateUntil(time.Second, csvPhaseSucceeded, ctx.Done())
+	err := wait.PollImmediateUntil(time.Second, csvPhaseSucceeded, ctx.Done())
+	if err != nil && errors.Is(err, context.DeadlineExceeded) {
+		depCheckErr := c.getDeploymentStatus(ctx, key, csv)
+		if depCheckErr != nil {
+			return fmt.Errorf("failed to run operator: %v %v", err, depCheckErr)
+		}
+	}
+
+	return err
+}
+
+// Get Deployment status
+
+func (c Client) getDeploymentStatus(ctx context.Context, key types.NamespacedName, csv olmapiv1alpha1.ClusterServiceVersion) error {
+	dep := &appsv1.Deployment{}
+	var depName string
+	for _, depSpec := range csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs {
+		if depSpec.Spec.Selector.MatchLabels["control-plane"] == "controller-manager" {
```

Comment:
This label is not required by OLM, so we shouldn't assume its presence.

Instead of choosing one of the deployments in the CSV, I think we need to iterate and apply the same check to all of them.

Reply:
If we do that, we might end up reporting deployment failures that may have happened in non-operator deployments.

Reply:
I am changing logic, to run through all deployments, and capture any failures.